crates/autopilot/src/run_loop.rs

```diff
@@ -147,6 +147,42 @@ impl RunLoop {
                 }
             };
 
+            {
+                // Also calculate the score from the solution itself
+                // For now, don't actually use it for ranking, instead log differences to solver
+                // provided score for debugging purposes.
+                let auction_prices = auction
+                    .prices
+                    .iter()
+                    .map(|(k, v)| {
+                        (
+                            (*k).into(),
+                            auction::Price::new(domain::eth::Ether(*v)).unwrap(),
+                        )
+                    })
+                    .collect();
+                let policies = auction
+                    .orders
+                    .iter()
+                    .map(|order| (order.uid, order.protocol_fees.clone()))
+                    .collect();
+                let settlement =
+                    domain::settlement::Settlement::from_solution(solution, &auction, auction_id)
+                        .map(|s| s.score(&auction_prices, &policies));
+
+                if let Ok(Ok(score)) = settlement {
+                    if score.0 != solution.score().get() {
```

Comment:
🙈 this is very painful to read.

Reply:
Hopefully less painful now that the `competition::Score` is implemented.